### ISSUE TITLE
Variable can be annotated with Final and ClassVar at the same time

### ISSUE THREAD

**Issue by uriyyo** [python/mypy#10477]
**Bug Report**

Variable can be annotated with `Final` and `ClassVar` at the same time.

But `PEP 591` says that this is forbidden:
>  Variables should not be annotated with both ClassVar and Final.

**To Reproduce**

1. Declarate class variable with both `Final` and `ClassVar` annotations.

```python
from typing_extensions import Final, ClassVar


class Foo:
    a: Final[ClassVar[int]] = 10
```

**Expected Behavior**

`mypy` should fail.

```
error: Variable should not be annotated with both ClassVar and Final
```

**Actual Behavior**

`mypy` do not raise any error. 

```
Success: no issues found in 1 source file
```

**Your Environment**

- Mypy version used: 0.812
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used: 3.6.8
- Operating system and version: macOS Big Sur 11.2.1
